### PR TITLE
Update Sphinx and Sphinx-AutoAPI

### PR DESCRIPTION
**Pull request by ashb**
I haven't looked in detail how much the output differs, so I am marking this for draft right now.

Spelling check might not pass, but the build was passing locally.

The words added to the spelling dictionary are because Sphinx now picks up typehints (yay!)

### CODE REVIEW

**Comment by potiuk** [comment]
:dancers:

**Comment by ashb** [comment]
Not sure what changed this time, maybe I never checked on v1.8.

File: airflow/models/dag.py [hunk 1]
```diff
@@ -2937,7 +2937,7 @@ def calculate_dagrun_date_fields(
 def dag(*dag_args, **dag_kwargs):
     """
     Python dag decorator. Wraps a function into an Airflow DAG.
-    Accepts kwargs for operator kwarg. Can be used to parametrize DAGs.
+    Accepts kwargs for operator kwarg. Can be used to parameterize DAGs.
```
Comment (uranusjr) [inline airflow/models/dag.py:2964]: *Parametrize* seems more widespread? That’s the spelling Pytest uses as well. I don’t know.

Comment (ashb) [reply airflow/models/dag.py:2964]: But not https://pypi.org/project/parameterized/ -- and this spelling was in the dictionary, the other wasn't.

(As a British English speaker param-et-er-ized is how I say it, so I was happy to go along with this.)

**Comment by uranusjr** [comment]
The biggest change (excluding changes caused by the underlying dependency upgrades, which isn’t visible in the source) is `exampleinclude` -> `literalinclude`. Any specific reason for those?

**Comment by ashb** [comment]
> The biggest change (excluding changes caused by the underlying dependency upgrades, which isn’t visible in the source) is `exampleinclude` -> `literalinclude`. Any specific reason for those?

exampleinclude was only ever designed to work with .py files, and sphinx got more specific about the "register_source" call we were doing there -- it errored on later Sphinx. literalinclude does what we want for those files, so it seemed easier to use that directive than change our exampleinclude to cope with non .py files

File: airflow/providers/google/cloud/utils/credentials_provider.py [hunk 2]
```diff
@@ -89,6 +89,8 @@ def provide_gcp_credentials(key_file_path: Optional[str] = None, key_file_dict:
     :type key_file_path: str
     :param key_file_dict: Dictionary with credentials.
     :type key_file_dict: Dict
+
+    __ https://cloud.google.com/docs/authentication/production
```
Comment (kaxil) [inline airflow/providers/google/cloud/utils/credentials_provider.py:93]: TIL -- didn't knew we could do this in docstrings too

**Review by kaxil — APPROVED** [review]

**Comment by kaxil** [comment]
Conflicts !

File: airflow/providers/google/cloud/utils/credentials_provider.py (hunk 2 above)
Comment (ashb) [reply airflow/providers/google/cloud/utils/credentials_provider.py:93]: Yup, Doc strings are "just" rst :)

**Comment by ashb** [comment]
Easy one to resolve -- it was TP's `:sphinx-autoapi-skip:' change that I'd already accounted for in `docs/conf.py` here

**Comment by ashb** [comment]
I might have a quick fix for the failing build, and I'd rather give this PR time for a few more eyes, that or someone to have looked At the built docs and say "yes, this output LGTM"

**Comment by ashb** [comment]
I might have a temporary fix for the failing build, and I'd rather give this PR time for a few more eyes, that or someone to have looked At the built docs and say "yes, this output LGTM"

**Comment by ashb** [comment]
I've  given this a cursor look, and it seems okay content wise -- possibly a few styling things we could fix up such as this 

Before:
image


After:
image

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
> I've given this a cursor look, and it seems okay content wise -- possibly a few styling things we could fix up such as this

I think we have quite a few other styling issues (most notably the cursed expanding-beyond-screen wide tables with vertical scrolling which make most of the wide tables hidden to a casual user)  so I think we should fix those in a separate PR. 

Maybe even someone from the UI team, could take a look because of course:

https://9elements.com/blog/content/images/2019/07/image-1.png

File: setup.py [hunk 3]
```diff
@@ -254,16 +254,16 @@ def write_version(filename: str = os.path.join(*[my_dir, "airflow", "git_version
 doc = [
     'click>=7.1,<9',
     # Sphinx is limited to < 3.5.0 because of https://github.com/sphinx-doc/sphinx/issues/8880
```
Comment (mik-laj) [inline setup.py:256]: Should we update the comment?

Comment (ashb) [reply setup.py:256]: Whoops, yes we should have.

**Pull request merged**